toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/internal/CalibrationState.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ *  Copyright 2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.ar.internal
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.setValue
+
+
+/**
+ * State holder class for calibration view
+ *
+ * @param onHeadingChange Lambda invoked when the user adjusts heading offset
+ * @param onHeadingReset Lambda invoked when the user resets heading offset
+ * @param onElevationChange Lambda invoked when the user adjusts elevation offset
+ * @param onElevationReset Lambda invoked when the user resets elevation offset
+ */
+@Immutable
```

Comment:
Having re-read the [doc](https://developer.android.com/develop/ui/compose/performance/stability#types), I think semantically it makes more sense to mark this with `@Stable`. In compose "stable" means "a type whose properties can change after construction. If and when those properties change during runtime, Compose becomes aware of those changes." - I think that's the case here.
```suggestion
@Stable
```

```
When applied to a class or an interface, Stable indicates that the following must be true:
- The result of equals will always return the same result for the same two instances.
- When a public property of the type changes, composition will be notified.
- All public property types are stable.
```